tasks/migrate-certificates.ts

```diff
@@ -0,0 +1,202 @@
+/* eslint-disable no-await-in-loop -- need to submit transactions synchronously to avoid nonce collisions */
+import cliProgress from 'cli-progress';
+import { task, types } from 'hardhat/config';
+import { BigNumber, ethers } from 'ethers';
+import chalk from 'chalk';
+import { parseTransactionLogs } from '@nori-dot-com/contracts/utils/events';
+import { readJsonSync, writeJsonSync } from 'fs-extra';
+
+import type { FireblocksSigner } from '@/plugins/fireblocks/fireblocks-signer';
+import { getCertificate, getRemoval } from '@/utils/contracts';
+
+interface MigrateCertificatesTaskOptions {
+  file: string;
+  outputFile?: string;
+  dryRun?: boolean;
+}
+
+type ParsedMigrateCertificatesTaskOptions = RequiredKeys<
+  MigrateCertificatesTaskOptions,
+  'file'
+>;
+
+export const GET_MIGRATE_CERTIFICATES_TASK = () =>
+  ({
+    name: 'migrate-certificates',
+    description: 'Utility to migrate legacy certificates',
+    run: async (
+      options: MigrateCertificatesTaskOptions,
+      _: CustomHardHatRuntimeEnvironment
+    ): Promise<void> => {
+      const {
+        file,
+        outputFile = 'migrated-certificates.json',
+        dryRun,
+      } = options as ParsedMigrateCertificatesTaskOptions;
+      const network = await hre.network.name;
+      if (![`localhost`, `mumbai`, `polygon`].includes(network)) {
+        throw new Error(
+          `Network ${network} is not supported. Please use localhost, mumbai, or polygon.`
+        );
+      }
+
+      const jsonData = readJsonSync(file);
+
+      const [signer] = await hre.getSigners();
+      const signerAddress = await signer.getAddress();
+      const removalContract = await getRemoval({
+        hre,
+        signer,
+      });
+      const certificateContract = await getCertificate({
+        hre,
+        signer,
+      });
+      hre.log(`Removal contract address: ${removalContract.address}`);
+      hre.log(`Certificate contract address: ${certificateContract.address}`);
+      hre.log(`Signer address: ${signerAddress}`);
+      // const fireblocksSigner = removalContract.signer as FireblocksSigner;
```

Comment:
I'd suggest using a similar technique to this one:

https://github.com/nori-dot-eco/contracts/blob/master/plugins/index.ts#L96

That will let you post memos on the transactions that the people signing the requests  will see.  Assume that's the only reason to care that it's a fireblocks signer?

i.e.

```
    const fireblocksSigner = signer as FireblocksSigner;
    if (typeof fireblocksSigner.setNextTransactionMemo === 'function') {
      fireblocksSigner.setNextTransactionMemo(
        `Migrate legacy certificates`
      );
    }
```